DataManagementSystem/Client/FileCatalogClientCLI.py

```diff
@@ -411,72 +429,106 @@ def removeFile(self,args):
         print(result['Message'])
     except Exception as x:
       print("Error: rm failed with exception: ", x)
-      
-  def removeDirectory(self,args):
-    """ Remove file from the catalog
-    """  
-    
-    path = args[0]
-    lfn = self.getPath(path)
-    print("lfn:", lfn)
+
+  def removeDirectory(self, lfn, recursive=False, forceNonEmpty=False):
+    """ Remove a given directory from the catalog. Remove multiple directories
+        recursively if recursive flag is True. Remove contained files if forceNonEmpty
+        flag is true.
+    """
+
+    if recursive or forceNonEmpty:
+      resultListDirectory = self.fc.listDirectory(lfn, False)
+      if not resultListDirectory['OK']:
+        print('Failed to look up the directory contents')
+        return S_ERROR('Failed to look up the directory contents')
+
+      # Remove subdirectories first
+      dirDict = resultListDirectory['Value']['Successful'][lfn]['SubDirs']
+      if dirDict:
+        if recursive:
+          for dirLfn in dirDict:
+            result = self.removeDirectory(dirLfn, recursive, forceNonEmpty)
+            if not result['OK']:
+              print('Error: failed to remove directory', dirLfn)
+              return S_ERROR('Failed to remove directory')
+        else:
+          print('Error: failed to remove non empty directory')
+          return S_ERROR("Failed to remove non empty directory")
+
+      # Remove files
+      fileDict = resultListDirectory['Value']['Successful'][lfn]['Files']
+      fileList = list(fileDict)
+      if fileList:
+        if forceNonEmpty:
+          print("Removing", len(fileList), "files in", lfn)
+          result = self.fc.removeFile(fileList)
```

Comment:
removeFile returns the Successful/Failed Structure (I am pretty sure). So you have to check also 'Failed'